src/bootstrap/compile.rs

```diff
@@ -1121,28 +1121,28 @@ pub fn run_cargo(builder: &Builder, cargo: &mut Command, stamp: &Path, is_check:
             CargoMessage::CompilerArtifact { filenames, .. } => filenames,
             _ => return,
         };
-        for filename in filenames {
+        for filename in filenames.iter() {
             // Skip files like executables
             if !filename.ends_with(".rlib") &&
                !filename.ends_with(".lib") &&
                !is_dylib(&filename) &&
                !(is_check && filename.ends_with(".rmeta")) {
-                return;
+                break;
             }
 
-            let filename = Path::new(&*filename);
+            let filename = Path::new(&**filename);
 
             // If this was an output file in the "host dir" we don't actually
             // worry about it, it's not relevant for us.
```

Comment:
Let's change this comment to basically refer to this same condition below and say see why there.